src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -122,66 +160,146 @@ trait GraphDAO {
     getSinglePipelineResult(entityPipeline(db, workspaceNamespace, workspaceName, entityType, entityName))
   }
 
-  import scala.reflect.runtime.universe._
-  import scala.reflect.runtime.{universe=>ru}
-  import scala.annotation.meta.field
-
   /**
-   * Takes the fields of obj annotated with VertexProperty and sets them on vertex.
-   * Vertex property keys are prefixed with an underscore.
+   * Serializes an object to a vertex. This supports all properties extending AnyVal, String, joda DateTime
+   * Map[String, Attribute], AttributeReference, Seq[Workflow], Seq[WorkflowFailure], MethodRepoConfiguration,
+   * MethodRepoMethod, WorkspaceName. Maps are stored in a sub vertex. Values of AttributeValue as properties on the
+   * sub vertex. Values of AttributeReference are stored as edges from the sub-vertex. Seq[Attribute] are stored as
+   * maps where the keys are the index.
    *
-   * @param obj source of data with which to update the vertex
-   * @param vertex the vertex to update
-   * @tparam T the type of obj
-   * @return the updated vertex
+   * @param obj to save
+   * @param vertex to save it to
+   * @param graph
+   * @param workspaceName used to populate workspace name fields
+   * @tparam T type of object being saved
+   * @return resulting vertex
    */
-  def setVertexProperties[T: TypeTag: ClassTag](obj: T, vertex: Vertex): Vertex = {
+  def saveToVertex[T: TypeTag: ClassTag](obj: T, vertex: Vertex, graph: Graph, workspaceName: WorkspaceName): Vertex = {
 
-    def getProperties(obj: T, prefix: String): Iterable[(String, Option[Any])] = {
+    def getProperties(obj: T): Iterable[(String, Option[Any])] = {
       val mirror = ru.runtimeMirror(obj.getClass.getClassLoader)
 
-      for (member <- ru.typeTag[T].tpe.members
-           if (member.asTerm.isVal || member.asTerm.isVar) && member.annotations.exists(_.tree.tpe.typeSymbol == ru.typeOf[VertexProperty@field].typeSymbol)
+      for (member <- ru.typeTag[T].tpe.members if (member.asTerm.isVal || member.asTerm.isVar)
       ) yield {
         val fieldMirror = mirror.reflect(obj).reflectField(member.asTerm)
-        prefix + member.name.decodedName.toString.trim -> (fieldMirror.get match {
+        member.name.decodedName.toString.trim -> (fieldMirror.get match {
           case v: Option[Any] => v
           case v => Option(v)
         })
       }
     }
 
-    getProperties(obj, "_").foreach(_ match {
-      case (key, None) => vertex.removeProperty(key)
+    getProperties(obj).foreach(_ match {
```

Comment:
point taken but I am going to leave it because I think this matches how I think about it better... we are not actually collecting anything, there is no result. We are performing a side effect for each...